### PR TITLE
Add Android.bp for 22.5.2

### PR DESCRIPTION
**Pull request by JeevakaPrabu**
Test: make libgmm_umd for x86_64

Change-Id: Ib89115762b31dfe1002a276b7cb409d4998af767

### CODE REVIEW

**Review by theandi666 — APPROVED** [review]
LGTM

File: Android.bp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        "-march=corei7",
+        "-fvisibility=hidden",
+        "-fno-omit-frame-pointer",
+        "-pthread",
+    ],
+
+    cppflags: [
+        "-Wno-implicit-fallthrough",
+        "-Wno-missing-braces",
+        "-Wno-parentheses-equality",
```
Comment (andrescj-chromium) [inline Android.bp:90]: I tried building without the following flags that I didn't find in the current system, and it works. Consider removing them from here:

- `-Wno-parentheses-equality`
- `-Wno-logical-not-parentheses`
- `-Wno-missing-field-initializers`

**Review by andrescj-chromium — CHANGES_REQUESTED** [review]
Looks good to me with changes. Thanks!

File: Android.bp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        "-Wno-logical-not-parentheses",
+        "-Wno-missing-field-initializers",
+        "-Wno-unknown-pragmas",
```
[the remark this answers is not in the document]
Comment (JeevakaPrabu) [reply Android.bp:93]: no issues seen even when not present in cflags

File: Android.bp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        "-D__UMD",
+        "-march=corei7",
+        "-fvisibility=hidden",
+        "-fno-omit-frame-pointer",
+        "-pthread",
+    ],
+
+    cppflags: [
+        "-Wno-implicit-fallthrough",
+        "-Wno-missing-braces",
```
[the remark this answers is not in the document]
Comment (JeevakaPrabu) [reply Android.bp:89]: no issues seen even when not present in cflags

File: Android.bp [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        "-D_X64",
+        "-D__GFX_MACRO_C__",
+        "-D__GMM",
+        "-D__STDC_CONSTANT_MACROS",
+        "-D__STDC_LIMIT_MACROS",
+        "-D__UMD",
+        "-march=corei7",
+        "-fvisibility=hidden",
+        "-fno-omit-frame-pointer",
+        "-pthread",
```
[the remark this answers is not in the document]
Comment (JeevakaPrabu) [reply Android.bp:84]: will move it to cppflags

**Review by andrescj-chromium — APPROVED** [review]
Looks good to me. Also tested locally for aosp_cf_x86_64_phone. Thanks!

**Comment by johnmach** [comment]
Merged in c743f06b430fab6debcb9e0bf82131636e4ede7b

**Pull request closed without merging**